src/github-access.tsx

```diff
@@ -0,0 +1,99 @@
+import * as http from 'http';
+import * as url from 'url';
+import * as open from 'open';
+import { saveEnvVariable } from './variables';
+import { fetch } from 'cross-fetch';
+
+const GITHUB_ACCESS_TOKEN = 'Github__OAuthAccessToken';
+
+type OAuthReturn = {
+  access_token: string;
+  scope: string;
+  token_type: string;
+};
+
+export class GithubAccess {
+  // "avensia-oss-garn" oauth app
+  private static OAuthClientId = '825e1cb8ef39aa07f127';
+  // This is safe to store in version control
+  private static OAuthClientSecret = 'a7e9440c087ad4410dc4f77485622cc7a17feda3';
+  // This port is important. It's defined as a valid callback url in the app settings in github
+  private static capturePort = 9437;
+
+  public static get accessToken() {
+    return process.env[GITHUB_ACCESS_TOKEN] || null;
+  }
+
+  public static set accessToken(token: string | null) {
+    process.env[GITHUB_ACCESS_TOKEN] = token ?? '';
+    saveEnvVariable(GITHUB_ACCESS_TOKEN, token ?? '');
+  }
+
+  public static hasCredentials(): boolean {
+    return !!GithubAccess.accessToken;
+  }
+
+  public static async validateCredentials() {
+    if (!GithubAccess.hasCredentials()) {
+      try {
+        await GithubAccess.captureCredentials();
+      } catch (e) {
+        throw new Error(`Failed to capture github credentials: ${e}`);
+      }
+    }
+  }
+
+  private static async captureCredentials() {
+    const promise = new Promise((resolve, reject) => {
+      const nonce = String(Math.random()).split('.')[1];
```

Comment:
you should use a cryptographically secure random source here.

Reply:
Sure